drivers/charger/sbs_charger/emul_sbs_charger.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2023 Cirrus Logic, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Emulator for SBS 1.1 compliant smart battery charger.
+ */
+
+#define DT_DRV_COMPAT sbs_sbs_charger
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(sbs_sbs_charger);
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/emul.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/i2c_emul.h>
+#include <zephyr/sys/byteorder.h>
+
+#include "sbs_charger.h"
+
+/** Static configuration for the emulator */
+struct sbs_charger_emul_cfg {
+	/** I2C address of emulator */
+	uint16_t addr;
+};
+
+static int emul_sbs_charger_reg_write(const struct emul *target, int reg, int val)
+{
+	LOG_INF("write %x = %x", reg, val);
+	switch (reg) {
+	default:
+		LOG_INF("Unknown write %x", reg);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int emul_sbs_charger_reg_read(const struct emul *target, int reg, int *val)
+{
+	switch (reg) {
+	case SBS_CHARGER_REG_SPEC_INFO:
+	case SBS_CHARGER_REG_STATUS:
+	case SBS_CHARGER_REG_ALARM_WARNING:
+		/* Arbitrary stub value. */
+		*val = 1;
+		break;
+	default:
+		LOG_ERR("Unknown register 0x%x read", reg);
+		return -EIO;
+	}
+	LOG_INF("read 0x%x = 0x%x", reg, *val);
+
+	return 0;
+}
+
+static int sbs_charger_emul_transfer_i2c(const struct emul *target, struct i2c_msg *msgs,
+					 int num_msgs, int addr)
+{
+	/* Largely copied from emul_bmi160.c */
```

Comment:
Should probably change to largely copied from emul_sbs_gauge ;)

Reply:
lol i <3 open source. Ack